File: backend/app/middlewares/rate_limiter.py
"""
rate_limiter.py — Rate limiting (SlowAPI / IP)
Responsable : Dev 1 (Senior - Anti-fraude)
"""
from slowapi import Limiter
from slowapi.util import get_remote_address

# TODO Dev 1 : clé composite poll_id + IP, fenêtres dynamiques
limiter = Limiter(key_func=get_remote_address)

File: backend/src/middlewares/rateLimiter.js
/**
 * rateLimiter.js — Middleware IP rate limiting
 * Responsable : Dev 1 (Senior - Anti-fraude)
 * Limite : 1 vote par IP par sondage (à affiner par poll_id)
 */
import rateLimit from 'express-rate-limit';

// TODO Dev 1 : configurer les fenêtres de temps et les messages d’erreur
export const voteRateLimiter = rateLimit({
  windowMs: 15 * 60 * 1000,
  max: 1,
  // TODO : personnaliser le message de refus
});
